src/runtime/inline/list.cpp

```diff
@@ -39,21 +39,42 @@ Box* listiterHasnext(Box* s) {
     assert(s->cls == list_iterator_cls);
     BoxedListIterator* self = static_cast<BoxedListIterator*>(s);
 
-    return boxBool(self->pos < self->l->size);
+    if (!self->l) {
+        raiseExcHelper(StopIteration, "");
```

Comment:
Could we return false here and rely on the caller to throw the StopIteration if they want one?